mikupad.html

```diff
@@ -1656,31 +1837,94 @@
+			// default to 2048
+			const searchRange = isNaN(entry.search) || entry.search === ""
+				? 2048
+				: Number(entry.search);
+			
+			// truncate to search range. using promptText allows for search ranges larger than context
+			const searchPrompt = promptText.substring(promptText.length - searchRange * tokenRatio);
+			
+			// search in range
+			return entry.keys.some((key, index) => {
+				// don't waste resources on disabled entries
+				if (searchPrompt.length == 0) {
+					return
+				}
+				
+				// an invalid regex here can completely lock you out of mikupad until you clear
+				// localStorage, so this is necessary to handle that.
+				try {
+					return new RegExp(key, "i").test(searchPrompt) && key !== "";
+				}
+				catch (error) {
+					console.error(`Error in RegEx for key '${key}': ${error.message}`);
+					return false;
+				}
+			});
+			
+		});
+		
+		const assembledWorldInfo = activeWorldInfo.length > 0
+			? activeWorldInfo.map(entry => entry.text).join("\n")
+			: "";
 
+		return assembledWorldInfo
+	}, [worldInfo]);
 
 	const modifiedPrompt = useMemo(() => {
-		// assemble individual contexts if text not empty
+		// add world info to memory for easier assembly
+		memoryTokens["worldInfo"] = assembledWorldInfo;
+		
 		const order = ["prefix","text","suffix"]
-		const assembledMemory = memoryTokens.text && memoryTokens.text !== ""
-			? order.map(key => memoryTokens[key]).join("").replace(/\\n/g,'\n')
-			: "";
 		const assembledAuthorNote = authorNoteTokens.text && authorNoteTokens.text !== ""
 			? order.map(key => authorNoteTokens[key]).join("").replace(/\\n/g,'\n')
 			: "";
 
+				// replacements for the contextOrder string
+				const replacements = {
+						"{wiPrefix}": memoryTokens.worldInfo && memoryTokens.worldInfo !== ""
+								? worldInfo.prefix
+								: "", // wi prefix and suffix will be added whenever wi isn't empty
+						"{wiText}": memoryTokens.worldInfo,
+						"{wiSuffix}": memoryTokens.worldInfo && memoryTokens.worldInfo !== ""
+								? worldInfo.suffix
+								: "",
+
+						"{memPrefix}": memoryTokens.text && memoryTokens.text !== "" || memoryTokens.worldInfo !== ""
+								? memoryTokens.prefix
+								: "", // memory prefix and suffix will be added whenever memory or wi aren't empty
+						"{memText}": memoryTokens.text,
+						"{memSuffix}": memoryTokens.text && memoryTokens.text !== "" || memoryTokens.worldInfo !== ""
+								? memoryTokens.suffix
+								: "",
+				}
+		
 		// prompt length estimation
-		const tokenRatio = 3.3;
-		const additionalContext = (assembledMemory + assembledAuthorNote).length; // TODO add active WI
+		const additionalContext = (Object.values(replacements)
+			.filter(value => typeof value === 'string').join('')).length;
 		const estimatedContextStart = Math.round(
-			promptText.length - contextLength * tokenRatio + additionalContext / tokenRatio) + 1;
-
-		// truncate prompt
+			promptText.length - contextLength * tokenRatio + additionalContext) + 1;

```

Comment:
I think the `/ tokenRatio` isn't necessary here? We're working with characters for this variable, not tokens, right?